### PR TITLE
Added KCTWrapper and Wrapper classes for KIP*

### PR DESCRIPTION
**Pull request by aeharvlee**
## Proposed changes

This PR is for improving SDK usability of KCT Layer in [Common Architecture](https://kips.klaytn.com/KIPs/kip-34)

Improving SDK usability means providing users with a similar development experience
no matter what programming language the SDK is implemented in.

Each layer declared in Common Architecture should be accessed via `caver`.
KCT Layer in Common Architecture should be accessed via `caver.kct`.

In caver-js, all of static methods declared in `KIP*` classes are accessed via `caver.kct.kip*`
e.g., `caver.kct.kip7.detectInterface()`, `caver.kct.kip17.deploy()`, etc…

To make similar development experiences in caver-java, I added some wrapper classes.

**1. Add Wrappers for KIP7, KIP17, KIP37**
By adding wrapper classes for `KIP7`, `KIP17`, `KIP37`, we can use
the static methods defined in the KIP of that type through `caver.kct.kip*`.

Each wrapper classes has a `Caver` instance as a member variable.
This allows SDK users to call static methods defined in the `KIP*`
without passing a `Caver` type instance and `IWallet` type instance as method parameters.

This member variable caver should be injected from `KCTWrapper`.

**2. Add KCTWrapper and add this as a member variable of Caver class**
`KCTWrapper` contains wrapper classes of `KIP*` which have static methods.

By adding this class and adding this as a member variable of `Caver` class,
we can use static methods declared in the `KIP*` through `caver.kct.kip*`
e.g., `caver.kct.kip7.detectInterface()`

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-java/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-java)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     * Map<String, Boolean> result = KIP7.detectInterface();
+     * result.get(KIP7.INTERFACE_ID_IKIP7);
+     * result.get(KIP7.INTERFACE_ID_IKIP7_BURNABLE);
+     * result.get(KIP7.INTERFACE_ID_IKIP7_METADATA);
+     * result.get(KIP7.INTERFACE_ID_IKIP7_MINTABLE);
+     * result.get(KIP7.INTERFACE_ID_IKIP7_PAUSABLE);
+     * }</pre>
+     *
+     * @param contractAddress
+     * @return
```
Comment (sirano11) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:101]:
```suggestion
     * @return Map&lt;String, Boolean&gt;
```

Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:101]: @sirano11 
Thanks for your feedback.
I'll check whether there are other missing types at `@return` and if there are, I'll update missed types.

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+     * Creates a KIP17Wrapper instance
+     * @param caver A Caver instance
+     */
+    public KIP17Wrapper(Caver caver) {
+        this.caver = caver;
+    }
+
+    /**
+     * Deploy a KIP-17 contract.
+     * It must add deployer's keyring in caver.wallet.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java:50]:
```suggestion
     * The deployer's keyring should be existed in `caver.wallet`.
```

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+     * @throws InvocationTargetException
+     * @throws TransactionException
+     */
+    public KIP17 deploy(String deployer, String name, String symbol) throws NoSuchMethodException, IOException, InstantiationException, ClassNotFoundException, IllegalAccessException, InvocationTargetException, TransactionException {
+        return KIP17.deploy(this.caver, deployer, name, symbol, this.caver.getWallet());
+    }
+
+    /**
+     * Deploy KIP17 contract
+     * It must add deployer's keyring in caver.wallet.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java:69]:
```suggestion
     * The deployer's keyring should be existed in `caver.wallet`.
```

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+     * Creates a KIP7Wrapper instance
+     * @param caver A Caver instance
+     */
+    public KIP7Wrapper(Caver caver) {
+        this.caver = caver;
+    }
+
+    /**
+     * Deploy KIP-7 contract.
+     * It must add deployer's keyring in caver.wallet.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:51]:
```suggestion
     * The deployer's keyring should be existed in `caver.wallet`.
```

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+     * @throws InvocationTargetException
+     * @throws TransactionException
+     */
+    public KIP7 deploy(String deployer, String name, String symbol, int decimals, BigInteger initialSupply) throws NoSuchMethodException, IOException, InstantiationException, ClassNotFoundException, IllegalAccessException, InvocationTargetException, TransactionException {
+        return KIP7.deploy(this.caver, deployer, name, symbol, decimals, initialSupply, this.caver.getWallet());
+    }
+
+    /**
+     * Deploy KIP-7 contract.
+     * It must add deployer's keyring in caver.wallet.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:72]:
```suggestion
     * The deployer's keyring should be existed in `caver.wallet`.
```

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    private Caver caver;
+
+    /**
+     * Creates a KIP17Wrapper instance
+     * @param caver A Caver instance
+     */
+    public KIP17Wrapper(Caver caver) {
+        this.caver = caver;
+    }
+
```
Comment (sirano11) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java:47]: @aeharvlee Don't we need to add a create()?

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP37Wrapper.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    private Caver caver;
+
+    /**
+     * Creates a KIP37Wrapper instance
+     * @param caver A Caver instance
+     */
+    public KIP37Wrapper(Caver caver) {
+        this.caver = caver;
+    }
+
```
Comment (sirano11) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP37Wrapper.java:46]: @aeharvlee Don't we need to add a create()?

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    private Caver caver;
+
+    /**
+     * Creates a KIP7Wrapper instance
+     * @param caver A Caver instance
+     */
+    public KIP7Wrapper(Caver caver) {
+        this.caver = caver;
+    }
+
```
Comment (sirano11) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:48]: @aeharvlee Don't we need to add a create()?

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java (hunk 2 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP17Wrapper.java:50]: @jimni1222 
Thanks for your feedback.
I'll update comments of KIP*Wrappers.

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java (hunk 1 above)
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:101]:
```suggestion
     * @return Map<String, Boolean>
```
Can you fill return type of those `detectInterface` function?

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java (hunk 8 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:48]: @sirano11 
I checked the [KCT Layer in Common Architecture](https://kips.klaytn.com/KIPs/kip-34#contract-abi-kct-layer) and yeah there was `create()`.
I think we need to add `create()` for KIP7, KIP17, and KIP37.
Sorry to missed it T_T.

I'll add implementation about it.
Thanks for your feedback.

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java (hunk 1 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:101]: @jimni1222 
Of course.
I'll :) 

Thanks for your feedback.

**Review by jimni1222 — APPROVED** [review]
LGTM =)!!!

File: core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java (hunk 8 above)
Comment (aeharvlee) [reply core/src/main/java/com/klaytn/caver/kct/kip_wrapper/KIP7Wrapper.java:48]: @sirano11 
Hi, I added `create()` methods for `KIP*Wrapper` classes based on your feedback and [KCT Layer Reference](https://kips.klaytn.com/KIPs/kip-34#contract-abi-kct-layer).

Here is the related commit :)
[Add create methods in KIP*Wrapper](https://github.com/klaytn/caver-java/pull/223/commits/0ca485fc7100564e6bea7324de7bc3243d43b498)

Thanks for your feedback again.

**Review by sirano11 — APPROVED** [review]

**Pull request merged**